src/force/CabanaPD_Force_PMB.hpp

```diff
@@ -0,0 +1,387 @@
+                {
+                    const double coeff = c * s * vol( j );
+                    double muij = mu( i, n );
+                    fx_i = muij * coeff * rx / r;
+                    fy_i = muij * coeff * ry / r;
+                    fz_i = muij * coeff * rz / r;
+
+                    f( i, 0 ) += fx_i;
+                    f( i, 1 ) += fy_i;
+                    f( i, 2 ) += fz_i;
+                }
+            }
+        };
+
+        Kokkos::RangePolicy<exec_space> policy( 0, n_local );
+        Kokkos::parallel_for( "CabanaPD::ForcePMBDamage::computeFull", policy,
+                              force_full );
+    }
+
+    template <class PosType, class WType, class DamageType, class ParticleType,
+              class NeighListType, class MuView, class ParallelType>
+    double computeEnergyFull( WType& W, const PosType& x, const PosType& u,
+                              DamageType& phi, const ParticleType& particles,
+                              const NeighListType& neigh_list, MuView& mu,
+                              const int n_local, ParallelType& ) const
+    {
+        auto c = _model.c;
+        const auto vol = particles.sliceVolume();
+
+        auto energy_full = KOKKOS_LAMBDA( const int i, double& Phi )
+        {
+            std::size_t num_neighbors =
+                Cabana::NeighborList<NeighListType>::numNeighbor( neigh_list,
+                                                                  i );
+            double phi_i = 0.0;
+            double vol_H_i = 0.0;
+            for ( std::size_t n = 0; n < num_neighbors; n++ )
+            {
+                std::size_t j =
+                    Cabana::NeighborList<NeighListType>::getNeighbor(
+                        neigh_list, i, n );
+                // Get the bond distance, displacement, and stretch.
+                double xi, r, s;
+                getDistance( x, u, i, j, xi, r, s );
+
+                // 0.25 factor is due to 1/2 from outside the integral and 1/2
+                // from the integrand (pairwise potential).
+                double w = mu( i, n ) * 0.25 * c * s * s * xi * vol( j );
+                W( i ) += w;
+
+                phi_i += mu( i, n ) * vol( j );
+                vol_H_i += vol( j );
+            }
+            Phi += W( i ) * vol( i );
+            phi( i ) = 1 - phi_i / vol_H_i;
+        };
+
+        double strain_energy = 0.0;
+        Kokkos::RangePolicy<exec_space> policy( 0, n_local );
+        Kokkos::parallel_reduce( "CabanaPD::ForcePMBDamage::computeEnergyFull",
+                                 policy, energy_full, strain_energy );
+
+        return strain_energy;
+    }
+};
+
+template <class ExecutionSpace>
+class Force<ExecutionSpace, ForceModel<LinearPMB, Elastic>>
+    : public Force<ExecutionSpace, ForceModel<PMB, Elastic>>
+{
+  protected:
+    using base_type = Force<ExecutionSpace, ForceModel<PMB, Elastic>>;
+    using base_type::_half_neigh;
+    ForceModel<LinearPMB, Elastic> _model;
```

Comment:
@streeve Is this line OK? What does it do?

Reply:
I don't understand the question - this stores the model..

Reply:
OK